client/src/pages/profile.tsx

```diff
@@ -49,31 +52,23 @@ const useStyles = makeStyles(() => ({
 
 function ProfilePage(props: { accessToken: string }): JSX.Element {
   const classes = useStyles();
-  const {
-    editedData: editedMentor,
-    error: mentorError,
-    isLoading: isMentorLoading,
-    isSaving: isMentorSaving,
-    isEdited: isMentorEdited,
-    editData: editMentor,
-    saveMentorDetails,
-  } = useWithMentor(props.accessToken);
+  const { state, saveMentor, editMentor } = useWithMentor(props.accessToken);
 
-  if (!editedMentor) {
+  if (!state.editedData) {
     return <div />;
   }
 
   return (
     <div className={classes.root}>
-      <NavBar title="My Profile" mentorId={editedMentor?._id} />
+      <NavBar title="My Profile" mentorId={state.editedData?._id} />
       <Paper className={classes.paper}>
         <div className={classes.inputField}>
           <FormControl>
             <InputLabel>Mentor Type</InputLabel>
             <Select
               data-cy="select-chat-type"
               label="Mentor Type"
-              value={editedMentor?.mentorType}
+              value={state.editedData?.mentorType}
```

Comment:
narrowing above

```typescript
if (!state.editedData) {
  return <div />;
}
```

...means that `state.editedData` CANNOT be undefined here. This is not a compile-time check; this is true at runtime. Make sure to always take proper advantage of narrowing. If you DON'T do the narrowing, you're not actually making you code more safe. Instead, you're making it more confusing/harder to read